tests/js/spec/views/issueList/overview.spec.jsx

```diff
@@ -674,355 +605,637 @@ describe('IssueList', function () {
         body: {
           ...savedSearch,
           isPinned: true,
         },
       });
+      const deletePin = MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/pinned-searches/',
+        method: 'DELETE',
+      });
 
-      wrapper.find('SavedSearchTab DropdownMenu a').simulate('click');
-      wrapper.find('SavedSearchMenuItem a').first().simulate('click');
-
-      await tick();
-
-      expect(browserHistory.push).toHaveBeenLastCalledWith(
-        expect.objectContaining({
-          pathname: '/organizations/org-slug/issues/searches/789/',
+      savedSearchesRequest = MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/searches/',
+        body: [savedSearch, assignedToMe],
+      });
+      createWrapper({
+        location: {
+          ...router.location,
           query: {
-            environment: [],
-            project: ['3559'],
-            statsPeriod: '14d',
-            sort: 'date',
+            query: 'assigned:me level:fatal',
           },
-        })
-      );
-
-      wrapper.setProps({
-        params: {
-          ...router.params,
-          searchId: '789',
         },
-      });
-
-      expect(getSavedSearchTitle(wrapper)).toBe('Unresolved TypeErrors');
-
-      wrapper.find('Button[aria-label="Pin this search"] button').simulate('click');
-
-      expect(createPin).toHaveBeenCalled();
-
-      await tick();
-      wrapper.update();
-
-      expect(browserHistory.push).toHaveBeenLastCalledWith(
-        expect.objectContaining({
-          pathname: '/organizations/org-slug/issues/searches/789/',
-        })
-      );
-
-      wrapper.setProps({
         params: {
           ...router.params,
           searchId: '789',
         },
       });
 
-      await tick();
-      wrapper.update();
-
-      expect(getSavedSearchTitle(wrapper)).toBe('Unresolved TypeErrors');
+      await waitFor(() => {
+        expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+      });
 
-      // Select other saved search
-      wrapper.find('SavedSearchTab DropdownMenu a').simulate('click');
-      wrapper.find('SavedSearchMenuItem a').last().simulate('click');
+      fireEvent.click(wrapper.getByTestId('saved-search-title'));
+      fireEvent.click(wrapper.getAllByText('Unresolved TypeErrors')[1]);
 
       expect(browserHistory.push).toHaveBeenLastCalledWith(
         expect.objectContaining({
-          pathname: '/organizations/org-slug/issues/searches/234/',
+          pathname: '/organizations/org-slug/issues/searches/789/',
           query: {
-            project: [],
             environment: [],
+            project: ['3559'],
             statsPeriod: '14d',
             sort: 'date',
           },
         })
       );
 
-      wrapper.setProps({
-        params: {
-          ...router.params,
-          searchId: '234',
-        },
-      });
+      expect(wrapper.getByTestId('saved-search-title')).toHaveTextContent(
+        'Unresolved TypeErrors'
+      );
```

Comment:
```suggestion
expect(wrapper.queryByText('Saved Searches')).toBe(null);
      expect(wrapper.getByText('Unresolved TypeErrors')).toBeInTheDocument(); 
```

not sure if we will get these strings but we could only use Text queries here